test/common/trafficmanager/fakeprovider/endpoint.go

```diff
@@ -26,12 +26,12 @@ const (
 )
 
 // NewEndpointsClient creates a client which talks to a fake endpoint server.
-func NewEndpointsClient(subscriptionID string) (*armtrafficmanager.EndpointsClient, error) {
+func NewEndpointsClient() (*armtrafficmanager.EndpointsClient, error) {
 	fakeServer := fake.EndpointsServer{
 		Delete:         EndpointDelete,
 		CreateOrUpdate: EndpointCreateOrUpdate,
 	}
-	clientFactory, err := armtrafficmanager.NewClientFactory(subscriptionID, &azcorefake.TokenCredential{},
+	clientFactory, err := armtrafficmanager.NewClientFactory(DefaultSubscriptionID, &azcorefake.TokenCredential{},
```

Comment:
why removing the parameter? it seems that it reduced the flexibility of the fake client?